vm-virtio/src/transport/pci_device.rs

```diff
@@ -307,17 +324,43 @@ impl VirtioPciDevice {
             .add_capability(&configuration_cap)
             .map_err(PciDeviceError::CapabilitiesSetup)?;
 
+        let msix_cap = MsixCap::new(
+            settings_bar,
+            self.msix_num,
+            MSIX_TABLE_BAR_OFFSET as u32,
+            MSIX_PBA_BAR_OFFSET as u32,
+        );
+        self.configuration
+            .add_capability(&msix_cap)
+            .map_err(PciDeviceError::CapabilitiesSetup)?;
+
         self.settings_bar = settings_bar;
         Ok(())
     }
 }
 
 impl PciDevice for VirtioPciDevice {
-    fn assign_irq(&mut self, irq_evt: EventFd, irq_num: u32, irq_pin: PciInterruptPin) {
+    fn assign_irq(
+        &mut self,
+        irq_evt: Arc<Box<Fn(u16) + Send + Sync>>,
+        irq_num: u32,
+        irq_pin: PciInterruptPin,
+    ) {
         self.configuration.set_irq(irq_num as u8, irq_pin);
         self.interrupt_evt = Some(irq_evt);
```

Comment:
How am I going to write on my `EventFd` when I'm not using MSI?

Reply:
Just pass a callback that does it.